bigtable/type_test.go

```diff
@@ -37,12 +39,54 @@ func aggregateProto() *btapb.Type {
 	}
 }
 
+func TestUnknown(t *testing.T) {
+	unsupportedType := &btapb.Type{
+		Kind: &btapb.Type_Float64Type{
+			Float64Type: &btapb.Type_Float64{},
+		},
+	}
+	got, ok := ProtoToType(unsupportedType).(unknown[btapb.Type])
+	if !ok {
+		t.Errorf("got: %T, wanted unknown[btapb.Type]", got)
+	}
+	gotJSON, err := MarshalJSON(got)
+	if err != nil {
+		t.Fatalf("Error calling MarshalJSON: %v", err)
+	}
+	result, err := UnmarshalJSON(gotJSON)
+	if err != nil {
+		t.Fatalf("Error calling UnmarshalJSON: %v", err)
+	}
+	if diff := cmp.Diff(result, got, cmpopts.IgnoreUnexported(unknown[btapb.Type]{})); diff != "" {
+		t.Errorf("Unexpected diff: \n%s", diff)
+	}
+	if !Equal(result, got) {
+		t.Errorf("Unexpected result. Got %#v, want %#v", result, got)
+	}
+}
+
 func TestInt64Proto(t *testing.T) {
 	want := aggregateProto()
-	got := Int64Type{}.proto()
+	it := Int64Type{Encoding: BigEndianBytesEncoding{}}
+	got := it.proto()
 	if !proto.Equal(got, want) {
 		t.Errorf("got type %v, want: %v", got, want)
 	}
+
+	gotJSON, err := MarshalJSON(it)
+	if err != nil {
+		t.Fatalf("Error calling MarshalJSON: %v", err)
+	}
```

Comment:
I think this could be pulled out into a helper method? it's basically the same for each type right?